pandas/tests/groupby/test_groupby.py

```diff
@@ -233,6 +233,39 @@ def test_apply_issues(self):
             lambda x: x['time'][x['value'].idxmax()])
         assert_series_equal(result, expected)
 
+    def test_apply_trivial(self):
+        # GH 20066
+        def gen_one_df(start=0):
+            df_this = pd.DataFrame({'key': ['a', 'a', 'b', 'b', 'a'],
+                                    'data': [1.0, 2.0, 3.0, 4.0, 5.0]},
+                                   columns=['key', 'data'])[start:]
+            return df_this
+
+        df = gen_one_df()
+        expected = pd.concat([gen_one_df(1), gen_one_df(1)],
+                             axis=1, keys=['float64', 'object'])
+        result = df.groupby([str(x) for x in df.dtypes],
+                            axis=1).apply(lambda x: gen_one_df(1))
+
+        assert_frame_equal(result, expected)
+
+    @pytest.mark.xfail(reason="didn't work yet")
```

Comment:
can you mark the issue and provide some more explanation

Reply:
done in my last commit.